### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by amuraru**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        | no|
| New feature?    | no |
| API breaks?     | no |
| Deprecations?   | no |
| License         | Apache 2.0 |

### What's in this PR?

Added a new `update-go-deps` Makefile target to allow upgrading deps moving forward.
Upgraded the dependencies.

### Why?

Keep the dependencies up to date

### Checklist

- [x] Implementation tested

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -6,11 +6,11 @@ require (
 	emperror.dev/errors v0.8.0
 	github.com/banzaicloud/istio-client-go v0.0.11
 	github.com/imdario/mergo v0.3.12
-	github.com/jetstack/cert-manager v1.4.4
+	github.com/jetstack/cert-manager v1.6.1
```
Comment (adamantal) [inline api/go.mod:9]: > ⚠️ Following their deprecation in version 1.5, the cert-manager APIVersions v1alpha2, v1alpha3, and v1beta1 are no longer served.

According to the cert-manager v1.6.0 release description, they've removed the support for these API versions. Could you check that koperator does not use these versions?

Comment (amuraru) [reply api/go.mod:9]: Thanks for the note @adamantal - compile and tests pass but will do extra checks

Comment (amuraru) [reply api/go.mod:9]: hi @adamantal - I checked and only v1 version is used throughout the operator code

Comment (adamantal) [reply api/go.mod:9]: Thanks for checking it!

**Review by adamantal — APPROVED** [review]
LGTM. Thanks!

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**